stdlib/public/core/Sequence.swift

```diff
@@ -428,29 +427,20 @@ extension Sequence {
       return result
     }
 
-    var hitEnd = false
     var iterator = self.makeIterator()
-    while true {
-      guard let element = iterator.next() else {
-        hitEnd = true
-        break
-      }
+    while let element = iterator.next() {
       if try isSeparator(element) {
-        if !appendSubsequence() {
-          continue
-        }
-        if result.count == maxSplits {
-          break
-        }
-      } else {
-        subSequence.append(element)
```

Comment:
This line is missing from your version of this function. Therefore the `subSequence` will be empty.
Try the following snippet:
```Swift
let xs = AnySequence(1..<5)
xs.split { $0 == 3 }.map(Array.init) // results in []
xs.split(omittingEmptySubsequences: false) { $0 == 3 }.map(Array.init) // results in [[], []]
```

Please note that you should not be testing this change on instances of `Array` (like in `[1,2,3].split { $0 == 2 }`) because `Collection` protocol provides it's own implementation of `split` [here](https://github.com/apple/swift/blob/master/stdlib/public/core/Collection.swift#L1285).